### PR TITLE
fix: Allow searching relations by name

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/4489, and we'll need to update all dest plugins to use `GetAll`.
We could also pass an option to `Get`.

Maybe leave this open so others can review too

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: schema/table.go [hunk 1]
```diff
@@ -83,6 +83,20 @@ func (tt Tables) Get(name string) *Table {
 	return nil
 }
 
+// GetAll returns a table by name. Returns top-level tables and relations.
+func (tt Tables) GetAll(name string) *Table {
```
Comment (disq) [inline schema/table.go:87]: I would make this the default `Get` and add a `GetTopLevel` etc. that looks only in the top level list.

Comment (erezrokah) [reply schema/table.go:87]: Maybe both `GetTopLevel` and `GetAll`?

Comment (disq) [reply schema/table.go:87]: Keeping the `Get` name would keep compatibility across plugins, even though there's only a handful dest plugins (if not just PG) that's using it.

Comment (hermanschaaf) [reply schema/table.go:87]: `Get` (same as before) + `GetTopLevel` (new function) would get my vote

**Review by hermanschaaf — APPROVED** [review]

File: schema/table.go (hunk 1 above)
Comment (erezrokah) [reply schema/table.go:87]: Cool, I'm good with either. Let me update the PR

**Comment by erezrokah** [comment]
I'll go ahead and force merge @yevgenypats so we can un-break the SDK

**Pull request merged**